### PR TITLE
uar-960 oe address comparisons fix

### PR DESCRIPTION
**Pull request by AndreKramer7654**
### JIRA link

https://companieshouse.atlassian.net/browse/UAR-960

### Change description

Concatenate the premises and street in the two OE details addresses before comparison to lessen chances of changes being detected. The Public API does not break out the premises so currently a change request is always raised.

### Work checklist

- [ ] Tests added where applicable

### Merge instructions

We are committed to keeping commit history clean, consistent and linear. To achieve this, this commit should be structured as follows:

```
<type>[optional scope]: <description>
```

and contain the following structural elements:

- fix: a commit that patches a bug in your codebase (this correlates with PATCH in semantic versioning),
- feat: a commit that introduces a new feature to the codebase (this correlates with MINOR in semantic versioning),
- BREAKING CHANGE: a commit that has a footer `BREAKING CHANGE:` introduces a breaking API change (correlating with MAJOR in semantic versioning). A BREAKING CHANGE can be part of commits of any type,
- types other than `fix:` and `feat:` are allowed, for example `build:`, `chore:`, `ci:`, `docs:`, `style:`, `refactor:`, `perf:`, `test:`, and others,
- footers other than `BREAKING CHANGE: <description>` may be provided.

### CODE REVIEW

**Review by rafahop — APPROVED** [review]
Looks good to me

**Review by bantao-ch — APPROVED** [review]

File: src/main/java/uk/gov/companieshouse/overseasentitiesapi/model/updatesubmission/changelist/commonmodels/Address.java [hunk 1]
```diff
@@ -155,16 +155,26 @@ public boolean equals(Object o) {
...
+                        normalise(address.getPremisesAndStreet()))
                 && StringUtils.equalsIgnoreCase(normalise(area), normalise(address.area))
                 && StringUtils.equalsIgnoreCase(normalise(postTown), normalise(address.postTown))
                 && StringUtils.equalsIgnoreCase(normalise(region), normalise(address.region))
                 && StringUtils.equalsIgnoreCase(normalise(postCode), normalise(address.postCode))
                 && StringUtils.equalsIgnoreCase(normalise(country), normalise(address.country));
     }
 
+    private String getPremisesAndStreet() {
+        if (StringUtils.isEmpty(houseNameNum)) {
```
Comment (bbell-companieshouse) [inline src/main/java/uk/gov/companieshouse/overseasentitiesapi/model/updatesubmission/changelist/commonmodels/Address.java:169]: Could do this using StringBuilder class

```suggestion
       StringBuilder sb = new StringBuilder();
       sb.append(houseNameNum);
       sb.append(" " + street);
       return sb.toString().trim();
```

Comment (bbell-companieshouse) [reply src/main/java/uk/gov/companieshouse/overseasentitiesapi/model/updatesubmission/changelist/commonmodels/Address.java:169]: Not a big issue- just seen that the code has been merged already so all good! :)

Comment (AndreKramer7654) [reply src/main/java/uk/gov/companieshouse/overseasentitiesapi/model/updatesubmission/changelist/commonmodels/Address.java:169]: StringJoiner could have been used but as it's only two items. We need to follow this up with some re-work as we have address comparisons in two places.

**Pull request merged**